### PR TITLE
Fixed issues with variable reduction

### PR DESCRIPTION
**Pull request by thorulf4**

### CODE REVIEW

File: src/Expression.h [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #include "string"
 #include "algorithm"
 #include <map>
+#include <iostream>
```
Comment (NicklasHugoy) [inline src/Expression.h:7]: Unnecessary include?

File: src/Expression.h [hunk 2]
```diff
@@ -128,9 +142,14 @@ class VariableExpression : public Expression
         if(new_id != names.end()){
             id = new_id->second;
         }else{
-            names[id] = "v" + std::to_string(names.size());
+            names[id] = "v" + std::to_string(names.size() - 1);
+            id = names[id];
```
Comment (ZcortezZ) [inline src/Expression.h:146]: shouldnt this be added for the array id as well?

File: src/bit_cuda/bit_cuda_transformer.h [hunk 3]
```diff
@@ -61,7 +61,7 @@ class BitCudaTransformer : public CfgVisitor<LatticeType>
             
             for (int i = 1; i < node.arguments.size(); i++)
             {
-                last.next_transfer_index = add_transfer_function(node.formal_parameters[0], node.arguments[i]);
+                last.next_transfer_index = add_transfer_function(node.formal_parameters[i], node.arguments[i]);
```
Comment (ZcortezZ) [inline src/bit_cuda/bit_cuda_transformer.h:64]: shouldnt "i" start as 0 or is it intentional that we start at 1?

**Review by ZcortezZ — APPROVED** [review]

File: src/bit_cuda/bit_cuda_transformer.h (hunk 3 above)
Comment (NicklasHugoy) [reply src/bit_cuda/bit_cuda_transformer.h:64]: the 0 index is handled right before the loop

**Review by ZcortezZ — APPROVED** [review]

File: src/variable_reduction.h [hunk 4]
```diff
@@ -95,6 +100,7 @@ template<typename LatticeType>
 void reduce_variables(std::vector<std::shared_ptr<FunctionEntryNode<LatticeType>>>& entry_nodes) {
     VariableReducer<LatticeType> reducer({TAINT_VAR, RETURN_VAR});
     for(auto &entry : entry_nodes){
+        std::cout << entry->function_id << " - " << (intptr_t)entry.get() <<  "\n";
```
Comment (NicklasHugoy) [inline src/variable_reduction.h:103]: Remove debug print

**Review by NicklasHugoy — CHANGES_REQUESTED** [review]

**Review by NicklasHugoy — DISMISSED** [review]

**Review by ZcortezZ — APPROVED** [review]

**Review by ZcortezZ — APPROVED** [review]

**Review by NicklasHugoy — APPROVED** [review]

**Pull request merged**